File: app/src/main/java/org/d1scw0rld/bookbag/dto/Property.java
package org.d1scw0rld.bookbag.dto;

import android.support.annotation.NonNull;

public class Property
{
   public int    iFieldTypeID;
   public long   iID = 0;
   public String sValue = "";

   public Property(int iFieldTypeID)
   {
      this.iFieldTypeID = iFieldTypeID;
   }

   public Property(int iFieldTypeID, String sValue)
   {
      this.iFieldTypeID = iFieldTypeID;
      this.sValue = sValue;
   }

   public Property(long iID, int iFieldTypeID, String sValue)
   {
      this.iID = iID;
      this.iFieldTypeID = iFieldTypeID;
      this.sValue = sValue;
   }

   public void copy(Property f)
   {
      iID = f.iID;
      iFieldTypeID = f.iFieldTypeID;
      sValue = f.sValue;
   }

   @NonNull
   @Override
   public String toString()
   {
      return sValue;
   }


//   @Override
//   public String getValue()
//   {
//      return sValue;
//   }

   @Override
   public boolean equals(Object o)
   {
      Property f = (Property) o;
      return o != null && iID == f.iID && iFieldTypeID == f.iFieldTypeID && sValue.equalsIgnoreCase(f.sValue);
   }
}

File: app/src/main/java/org/d1scw0rld/bookbag/dto/Property.kt
package org.d1scw0rld.bookbag.dto;

import androidx.annotation.NonNull;

public class Property
{
   public int    fieldTypeId;
   public long   id = 0;
   public String value = "";

   public Property(int fieldTypeId)
   {
      this.fieldTypeId = fieldTypeId;
   }

   public Property(int fieldTypeId, String value)
   {
      this.fieldTypeId = fieldTypeId;
      this.value = value;
   }

   public Property(long id, int fieldTypeId, String value)
   {
      this.id = id;
      this.fieldTypeId = fieldTypeId;
      this.value = value;
   }

   public void copy(Property other)
   {
      id = other.id;
      fieldTypeId = other.fieldTypeId;
      value = other.value;
   }

   @NonNull
   @Override
   public String toString()
   {
      return value;
   }

   @Override
   public boolean equals(Object other)
   {
      if (this == other) return true;
      if (other == null || getClass() != other.getClass()) return false;
      Property property = (Property) other;
      return id == property.id && fieldTypeId == property.fieldTypeId && value.equalsIgnoreCase(property.value);
   }
}
